analyses/test_model_saving.ipynb

```diff
@@ -0,0 +1,337 @@
+       "HBox(children=(IntProgress(value=0, max=1500), HTML(value='')))"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "\n"
+     ]
+    }
+   ],
+   "source": [
+    "irm_results = classifier.train_with_irm(net, map_file, train_dirs, tune_dirs, gene_file, num_epochs, \n",
+    "                                        loss_scaling_factor, label_to_encoding, device, logger, '../logs/irm.pkl', 5)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Load the model and ensure the weights saved properly"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "trained_net = torch.load('../logs/irm.pkl')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sample_to_label = util.parse_map_file(map_file)\n",
+    "\n",
+    "tune_dataset = dataset.ExpressionDataset(tune_dirs, sample_to_label, label_to_encoding, gene_file)\n",
+    "tune_loader = torch.utils.data.DataLoader(tune_dataset, batch_size=16, num_workers=4, pin_memory=True)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Test trained network"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "tune_loss = 0\n",
+    "tune_correct = 0\n",
+    "\n",
+    "for tune_batch in tune_loader:\n",
+    "    expression, labels, ids = tune_batch                                            \n",
+    "    tune_expression = expression.to(device)                                         \n",
+    "    tune_labels = labels.to(device).double()                                        \n",
+    "\n",
+    "    loss_function = torch.nn.BCEWithLogitsLoss()                                          \n",
+    "\n",
+    "    tune_preds = trained_net(tune_expression)                                        \n",
+    "    loss = loss_function(tune_preds, tune_labels)\n",
+    "    \n",
+    "    tune_loss += float(loss)\n",
+    "    tune_correct += util.count_correct(tune_preds, tune_labels)\n",
+    "    \n",
+    "avg_loss = tune_loss / len(tune_dataset)\n",
+    "tune_acc = tune_correct / len(tune_dataset)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 12,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Trained network tune accuracy: 1.0\n",
+      "Trained network tune loss: 5.13992141030165e-19\n"
+     ]
+    }
+   ],
+   "source": [
+    "print('Trained network tune accuracy: {}'.format(tune_acc))\n",
+    "print('Trained network tune loss: {}'.format(avg_loss))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Test untrained network"
```

Comment:
The weights are initialized with the default PyTorch scheme, which I think is uniform with a range proportional to the square root of the number of weights in the layer.

I ran the untrained model as a negative control. If an untrained model had good performance at classifying the test data for some reason, it would mean that the loading the trained model wasn't necessarily successful.